sql/api/src/main/scala/org/apache/spark/sql/types/DataType.scala

```diff
@@ -117,6 +117,7 @@ object DataType {
   private val FIXED_DECIMAL = """decimal\(\s*(\d+)\s*,\s*(\-?\d+)\s*\)""".r
   private val CHAR_TYPE = """char\(\s*(\d+)\s*\)""".r
   private val VARCHAR_TYPE = """varchar\(\s*(\d+)\s*\)""".r
+  private val COLLATED_STRING_TYPE = """string\(\s*([\w_]+)\s*\)""".r
```

Comment:
since it's a DDL string, shall we follow the actual SQL syntax and use `STRING COLLATE name`?